main_test.go

```diff
@@ -0,0 +1,156 @@
+package main
```

Comment:
I don't see the place in the CI where those are executed, please make sure they at least run on every PR.